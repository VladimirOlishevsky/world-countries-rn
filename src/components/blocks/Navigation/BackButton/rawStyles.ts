import { StyleSheet } from "react-native";

export const makeStyles = (theme: ReactNativePaper.Theme) => StyleSheet.create({
  iconStyle: {
    alignItems: 'center',
    justifyContent: 'center',
    width: theme.spacing * 2.5,
    height: theme.spacing * 2.5,
  },
})
