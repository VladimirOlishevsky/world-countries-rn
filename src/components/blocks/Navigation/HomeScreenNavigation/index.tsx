import * as React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { Home } from '../../Home';
import { ContinentCountries } from '../../Home/ContinentCountries';
import { SeparatePage } from '../../../../sepa';
import { CountryComponent } from '../../Home/Country/index';
import { Maps } from '../../Home/Maps';
import { useTheme } from 'react-native-paper';

const Stack = createStackNavigator();

export const HomeScreenNavigation = () => {

  const theme = useTheme()

  return (
    <Stack.Navigator >
      <Stack.Screen name="HomePage" component={Home} options={{ headerShown: false }} />
      <Stack.Screen name="ContinentCountries" component={ContinentCountries} options={{ headerShown: false }} />
      <Stack.Screen name="Country" component={CountryComponent} options={{ headerShown: false }} />
      <Stack.Screen name="Maps" component={Maps} options={{
        title: 'Map',
        headerStyle: {
          backgroundColor: theme.colors.orange,
        },
        headerTintColor: theme.colors.white,
        headerTitleStyle: {
          fontWeight: 'bold',
        },
      }} />
      <Stack.Screen name="Test" component={SeparatePage} />
    </Stack.Navigator>
  );
}
