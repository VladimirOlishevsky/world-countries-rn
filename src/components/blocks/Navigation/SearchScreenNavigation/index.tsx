import * as React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { SearchScreen } from '../../SearchPage';
import { RegionalCountries } from '../../SearchPage/RegionalCountries';
import { Country } from '../../Home/Country';


const Stack = createStackNavigator();

export const SearchScreenNavigation = () => {
    return (
      <Stack.Navigator >
        <Stack.Screen name="SearchPage" component={SearchScreen} options={{ headerShown: false }}/>
        <Stack.Screen name="RegionalCountries" component={RegionalCountries} options={{ headerShown: false }}/>
        <Stack.Screen name="Country" component={Country} options={{ headerShown: false }}/>
      </Stack.Navigator>
    );
  }
