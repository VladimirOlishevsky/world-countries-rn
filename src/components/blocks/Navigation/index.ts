export * from './BackButton';
export * from './BottomNavigation';
export * from './HomeScreenNavigation';
export * from './SearchScreenNavigation';
