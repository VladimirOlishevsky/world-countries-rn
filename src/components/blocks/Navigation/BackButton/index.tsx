import * as React from 'react';
import { useNavigation } from '@react-navigation/native';
import { makeStyles } from './rawStyles';

import Icon from 'react-native-vector-icons/FontAwesome';
// import { TouchableOpacity } from 'react-native-gesture-handler';
import { useTheme } from 'react-native-paper';
import { TouchableOpacity } from 'react-native';

interface IProps {
    page?: string,
    testId?: string
}

export const BackButton = ({
    page,
    testId
}: IProps) => {
    const navigation = useNavigation();
    const theme = useTheme();
    const styles = makeStyles(theme)

    return (
        <TouchableOpacity
            testID={testId}
            onPress={() => page ? navigation.navigate(page) : navigation.goBack()}
            style={styles.iconStyle}
        >
            <Icon
                name="chevron-left"
                size={theme.size}
                color="#ff9f0a" />
        </TouchableOpacity>

    );
}
