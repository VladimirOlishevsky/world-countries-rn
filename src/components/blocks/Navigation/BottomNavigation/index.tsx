import * as React from 'react'
import { Image, StyleSheet } from 'react-native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { observer } from "mobx-react";
import { HomeScreenNavigation } from '../HomeScreenNavigation/index';
import { SearchScreenNavigation } from '../SearchScreenNavigation';
import { makeStyles } from './rawStyles';
import { useTheme } from 'react-native-paper';

const Tab = createBottomTabNavigator();

export const MainNavigation = observer(() => {

    const theme = useTheme()
    const styles = makeStyles(theme);
    
    return (
        <Tab.Navigator
            screenOptions={({ route }: any) => ({
                tabBarIcon: ({ focused, color, size }: any) => {
                    let iconName;
                    if (route.name === 'HomePage') {
                        focused ? iconName = require('../../../../../static/images/vicharanFocused.png')
                            : iconName = require('../../../../../static/images/vicharan.png');
                    }
                    else if (route.name === 'SearchPage') {
                        focused ? iconName = require('../../../../../static/images/kathaFocused.png')
                            : iconName = require('../../../../../static/images/katha.png');
                    }
                    return <Image source={iconName}
                        style={styles.image}
                    />;
                },
            })}
        >
            <Tab.Screen name="HomePage" component={HomeScreenNavigation} />
            <Tab.Screen name="SearchPage" component={SearchScreenNavigation} />
        </Tab.Navigator>
    )
})


