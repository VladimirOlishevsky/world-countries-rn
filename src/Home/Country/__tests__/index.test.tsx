import React from 'react';
import { Country } from '..';
import renderer from 'react-test-renderer';
import { BackButton } from 'Navigation/BackButton';
import { Button } from 'react-native-paper';
// import { Button } from 'native-base';


const mockFetchCountryByCode = jest.fn();
const mockStore = {
  countryStore: {
    name: '',
    fetchCountryByCode: mockFetchCountryByCode,
  },
};

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  },
  // ...props
});

describe('Country', () => {
  it('Country component render', () => {
    const push = jest.fn();
    const country = renderer.create(<Country navigation={{ push }} />).toJSON();
    // const country = render(<Country navigation={{ push }} />);
    expect(country).toMatchSnapshot();
  });
  it('button click', async () => {
    const navigate = 'regional';

    const country = renderer.create(<Country navigation={{ navigate }} />);
    const page = country.root.findByType(BackButton).props.page

    expect(page).toEqual('RegionalCountries');
  });
  it('countryStore name empty', async () => {
    // const navigate = 'regional';
    // let props: any;   
    // beforeEach(() => {
    //   props = createTestProps({});
    //   // wrapper = shallow(<LoadingScreen {...props} />);   // no compile-time error
    // });

    // const push = jest.fn();
    // const country = renderer.create(<Country navigation={() => {}} route={{ key: '1', name: '3' }} />);

    // const aa = country.root.findByType(Button).props as {
    //   onClick: () => void
    // };

    // console.log(aa)

    // const page = country.root.findByType(BackButton).props.page

    // expect(page).toEqual('RegionalCountries');
  });
});
