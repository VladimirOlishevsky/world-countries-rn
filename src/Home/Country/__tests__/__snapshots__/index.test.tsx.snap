// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen 1 Country component render 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "height": 1334,
      "padding": NaN,
      "paddingBottom": NaN,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": NaN,
        "marginTop": NaN,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": NaN,
          "justifyContent": "center",
          "opacity": 1,
          "width": NaN,
        }
      }
      testID="backButton"
    >
      <Icon
        color="#ff9f0a"
        name="chevron-left"
      />
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": NaN,
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
              "marginTop": NaN,
            }
          }
        >
          Country not found
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": undefined,
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": undefined,
              "elevation": 0,
              "marginTop": undefined,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#fff",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                go to search
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
